src/lib.rs

```diff
@@ -918,12 +918,17 @@ impl<'a, 'input: 'a> Node<'a, 'input> {
     /// assert_eq!(doc.root_element().namespaces().len(), 1);
     /// ```
     #[inline]
-    pub fn namespaces(&self) -> &'a [Namespace<'input>] {
+    pub fn namespaces(&self) -> Vec<&'a Namespace<'input>> {
```

Comment:
Sorry, but the public API must not allocate. Only parsing is allowed to allocate. That's the rule.

So this should be rewritten to iterator or something.

Reply:
Yeah, it was an unfortunate side effect of this PR. It could easily return an iterator instead, as you say, if that is acceptable.